math/dec_mut.go

```diff
@@ -0,0 +1,240 @@
+package math
+
+import "math/big"
+
+// LegacyDecMut is a wrapper around big.Int with an unsafe and mutable API
+type LegacyDecMut big.Int
```

Comment:
some problem with the approach is that it'd be now possible to create an overflowed dec by doing `LegacyDecMut(overflowedBigInt)`.

Is this a problem?